src/graph/on_disk_graph.cpp

```diff
@@ -10,52 +10,132 @@ using namespace kuzu::common;
+    // Empty columnIDs since we do not scan any rel property.
+    auto columnIDs = std::vector<column_id_t>{};
+    auto scanState = std::make_unique<RelTableScanState>(columnIDs, direction);
+    scanState->nodeIDVector = srcVector;
+    scanState->outputVectors.push_back(dstVector);
+    return scanState;
+}
+
+OnDiskGraphScanState::OnDiskGraphScanState(MemoryManager* mm) {
     srcNodeIDVectorState = DataChunkState::getSingleValueDataChunkState();
     dstNodeIDVectorState = std::make_shared<DataChunkState>();
     srcNodeIDVector = std::make_unique<ValueVector>(*LogicalType::INTERNAL_ID(), mm);
     srcNodeIDVector->state = srcNodeIDVectorState;
     dstNodeIDVector = std::make_unique<ValueVector>(*LogicalType::INTERNAL_ID(), mm);
     dstNodeIDVector->state = dstNodeIDVectorState;
-    fwdReadState = std::make_unique<RelTableScanState>(columnIDs, direction);
-    fwdReadState->nodeIDVector = srcNodeIDVector.get();
-    fwdReadState->outputVectors.push_back(dstNodeIDVector.get());
+    fwdScanState =
+        getRelScanState(RelDataDirection::FWD, srcNodeIDVector.get(), dstNodeIDVector.get());
+    bwdScanState =
+        getRelScanState(RelDataDirection::BWD, srcNodeIDVector.get(), dstNodeIDVector.get());
 }
 
-OnDiskGraph::OnDiskGraph(ClientContext* context, common::table_id_t nodeTableID,
-    common::table_id_t relTableID)
-    : context{context} {
+OnDiskGraph::OnDiskGraph(ClientContext* context, const GraphEntry& entry)
+    : transaction{context->getTx()}, graphEntry{entry.copy()},
+      scanState{context->getMemoryManager()} {
     auto storage = context->getStorageManager();
-    nodeTable = storage->getTable(nodeTableID)->ptrCast<NodeTable>();
-    relTable = storage->getTable(relTableID)->ptrCast<RelTable>();
-    nbrScanState = std::make_unique<NbrScanState>(context->getMemoryManager());
+    auto catalog = context->getCatalog();
+    for (auto& nodeTableID : graphEntry.nodeTableIDs) {
+        nodeIDToNodeTable.insert(
+            {nodeTableID, storage->getTable(nodeTableID)->ptrCast<NodeTable>()});
+        auto& nodeTableEntry = catalog->getTableCatalogEntry(transaction, nodeTableID)
+                                   ->constCast<NodeTableCatalogEntry>();
+        common::table_id_map_t<storage::RelTable*> fwdRelTables;
+        for (auto& relTableID : nodeTableEntry.getFwdRelTableIDSet()) {
+            if (!graphEntry.containsRelTableID(relTableID)) {
+                continue;
+            }
+            fwdRelTables.insert({relTableID, storage->getTable(relTableID)->ptrCast<RelTable>()});
+        }
+        nodeTableIDToFwdRelTables.insert({nodeTableID, std::move(fwdRelTables)});
+        common::table_id_map_t<storage::RelTable*> bwdRelTables;
+        for (auto& relTableID : nodeTableEntry.getBwdRelTableIDSet()) {
+            if (!graphEntry.containsRelTableID(relTableID)) {
+                continue;
+            }
+            bwdRelTables.insert({relTableID, storage->getTable(relTableID)->ptrCast<RelTable>()});
```

Comment:
When we develop the on-disk-graph, do we construct a version with both forward and backward edges? Many graph algorithms only need one direction.

If we have a TODO list for algorithms package, can you add "Support for constructing graphs (on-disk and in-mem) that only contain one directional relationships."